tests/bluetooth/bsim_bt/bsim_test_audio/src/tbs_client_test.c

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) 2019 Bose Corporation
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef CONFIG_BT_TBS_CLIENT
+
+#include <bluetooth/audio/tbs.h>
+
+#include "common.h"
+
+static struct bt_conn_cb conn_callbacks;
+extern enum bst_result_t bst_result;
+static volatile bool bt_init;
+static volatile bool is_connected;
+static volatile bool discovery_complete;
+static volatile bool is_gtbs_found;
+static volatile bool read_complete;
+static volatile bool call_placed;
+static volatile uint8_t call_state;
+static volatile uint8_t call_index;
+static struct bt_conn *tbs_conn;
+
+static void tbs_client_call_states_cb(struct bt_conn *conn, int err,
+				      uint8_t index, uint8_t call_count,
+				      const struct bt_tbs_client_call_state *call_states)
+{
+	if (index != 0) {
+		return;
+	}
+
+	printk("%s\n", __func__);
+	printk("Index %u\n", index);
+	if (err != 0) {
+		FAIL("Call could not read call states (%d)\n", err);
+		return;
+	}
+
+	call_index = call_states[0].index;
+	call_state = call_states[0].state;
+	printk("call index %u - state %u\n", call_index, call_state);
+}
+
+static void tbs_client_read_bearer_provider_name(struct bt_conn *conn, int err,
+						 uint8_t index,
+						 const char *value)
+{
+	if (err != 0) {
+		FAIL("Call could not read bearer name (%d)\n", err);
+		return;
+	}
+
+	printk("Index %u\n", index);
+	printk("Bearer name pointer: %p\n", value);
+	printk("Bearer name: %s\n", value);
+	read_complete = true;
+}
+
+static void tbs_client_discover_cb(struct bt_conn *conn, int err,
+				   uint8_t tbs_count, bool gtbs_found)
+{
+	printk("%s\n", __func__);
+	if (err != 0) {
+		FAIL("TBS_CLIENT could not be discovered (%d)\n", err);
+		return;
+	}
+
+	is_gtbs_found = true;
+	discovery_complete = true;
+}
+
+static const struct bt_tbs_client_cb tbs_client_cbs = {
+	.discover = tbs_client_discover_cb,
+	.originate_call = NULL,
+	.terminate_call = NULL,
+	.hold_call = NULL,
+	.accept_call = NULL,
+	.retrieve_call = NULL,
+	.join_calls = NULL,
+	.bearer_provider_name = tbs_client_read_bearer_provider_name,
+	.bearer_uci = NULL,
+	.technology = NULL,
+	.uri_list = NULL,
+	.signal_strength = NULL,
+	.signal_interval = NULL,
+	.current_calls = NULL,
+	.ccid = NULL,
+	.status_flags = NULL,
+	.call_uri = NULL,
+	.call_state = tbs_client_call_states_cb,
+	.termination_reason = NULL
+};
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (err != 0) {
+		FAIL("Failed to connect to %s (%u)\n", addr, err);
+		return;
+	}
+	printk("Connected to %s\n", addr);
+	tbs_conn = conn;
```

Comment:
Missing bt_conn_ref() here?

Reply:
Thanks for reminding me of applying the same fixes for these BSIM tests as I did for the others :) Remove the `tbs_conn` and updated to use the `default_conn` like the other LE Audio BSIM tests, as well as fixing the `bt_conn_ref`s.